### DISCUSSION TITLE
Default share theme

### DISCUSSION THREAD

**Discussion by zerebos** [TriliumNext/Trilium#4403] [Q&A]
## Background

With the 0.62 beta there is now a search endpoint available for the shared pages however the default share theme (and template) do not make use of it. I did that intentionally because I want to get input from the community before I made a PR to change the default share format.

## My Thoughts

My overall idea of what I think the default share should be is my website https://trilium.rocks/, it is a share theme and template I created specifically for that site. Here are the reasons I think it's a good default:

- The light and dark modes match the default Trilium themes and color schemes well.
- It has a functional search by default that only searches shared notes in the scope of the main tree.
- The collapsible trees make navigation compact and don't overwhelm users.
- The style and quality matches that of sharing from similar apps.
- There is complete mobile compatibility and optimization.
- It includes an active ToC.

But I know it has some downsides and decisions that would have to be made to make it more generally compatible.
- Notes that contain child notes would either have to act solely as "folders" and not have separate pages (like the site example), or the notes can be separate pages but users have to click the little arrow to expand sections instead of the whole label.
- Having a specialized default theme makes it more difficult for users that want to do their own themeing/customization.
- The style and structure (like ToC) may not fit other user-cases well.

## Your Thoughts

With all that said, for anyone that uses the share feature, I'd love to hear your thoughts on any of the following example questions, OR just your general wishlist for sharing/default share.

1. Should the default share format have a functional search bar?
2. Should the default share theme have a light/dark switch?
3. Should the default share page have collapsible tree sections?
4. Would you prefer if the default share stayed the same and instead there was a second built-in share theme that included those features above?
5. Generally, how would you like the default share to look?
6. What do you think of the design of https://trilium.rocks/? Would it make a good default share?

**Comment by meichthys** [comment]
My Thoughts:
1. Yes
2. Yes
3. Yes 
4. No
5. From a user perspective I would expect the share theme to match the theme that the user/author has selected, but I realize this could be difficult.
6. Trillium.rocks looks great to me. I did notice a few minor things:
- The table of contents does not scroll
- The search is absolutely amazing and clean (note labels and relations are hidden). Is this search based off of the default trillium search?
- As a Trilium user I would expect every label to be clickable with content. I think I would prefer to make the whole label clickable and require the user to click the down arrow to expand (especially if the "folder" note already has pills at the bottom showing the child notes)

**Reply by zerebos** [reply]
> From a user perspective I would expect the share theme to match the theme that the user/author has selected, but I realize this could be difficult.

It would be partially possible I would say, but yeah very difficult because themes have such free reign. Though I would intend for the share theme to be easily customizable with css variables just like Trilium itself.

> The table of contents does not scroll

I'm not sure what this means, it should change which one is active as you scroll the page and such

https://github.com/zadam/trilium/assets/6865942/46850b47-8db8-4725-9d97-654b4cb02a72

> Is this search based off of the default trillium search?

It is! You can take a look at my implementation here https://github.com/zadam/trilium/pull/4298/files#diff-0c186232b566b7b780da32b566e28cc06dca38fcde92d1cd5250a00639c1e7c2R334-R364

> As a Trilium user I would expect every label to be clickable with content. I think I would prefer to make the whole label clickable and require the user to click the down arrow to expand (especially if the "folder" note already has pills at the bottom showing the child notes)

I completely understand this and I think for Trilium users what you said is intuitive and understandable enough because it matches what actually happens in the app. I chose the other option for the site because in my mind a public share targets everyone including non-Trilium users, and I think making the whole label the collapsible section is more intuitive generally (and matches that of the share of apps similar to Trilium).

**Comment by Nyar233** [comment]
6. trilium.rocks theme looks great, this would be a good idea!

**Reply by fliptoback** [reply]
Second the trilium rocks theme - can we use this now? Really would like this implemented soonish.

**Reply by zerebos** [reply]
It currently depends on either template changes or custom templates (which were introduced in 0.62-beta), so in theory I could release it for those on the new beta

**Reply by fliptoback** [reply]
I am on the latest 0.61.13 not sure if this is a beta - but very keen to try out this new custom template.

**Reply by zerebos** [reply]
0.61.x is now the current stable release

**Comment by zadam** [comment]
As mentioned earlier, I really like your theme and would be very happy to accept a PR.

1. Yes
2. Yes
3. Yes
4. No, in the sense that I think this "batteries included" theme should be the default, but we should also provide an existing template which provides backwards compatibility and perhaps also serves as a more easily customizable base. But I think you can ignore this part in your PR and I will have to think about this ...

> Notes that contain child notes would either have to act solely as "folders" and not have separate pages (like the site example), or the notes can be separate pages but users have to click the little arrow to expand sections instead of the whole label.

This is the only thing which I understand as problematic, since it doesn't align with the Trilium's way of structuring content where there's no folder/file dichotomy. So I think here the little arrow would be preferable which would preserve the ability to click/activate also the folder.

**Reply by zerebos** [reply]
> But I think you can ignore this part in your PR and I will have to think about this

Alright, always willing to help brainstorm! I have a couple ideas.

> So I think here the little arrow would be preferable which would preserve the ability to click/activate also the folder.

Yeah I agree, I think for the PR for the default theme will use this method that matches Trilium.

**Reply by zerebos** [reply]
Hey @zadam I finally got my template to a good place where it can be switched between my version (entire menu item is collapsible) versus what would match Trilium (only arrow is collapsible) by just changing a couple lines of code.

That said, I know we talked in the past about whether this would be a "new feature" as far as maintenance mode goes and you seemed open to still including this. We also talked about how there should still be a "plain" share theme that is very basic. So, if you still want this to be added to current Trilium, how would you like it added? I could:

- make a PR that just replaces the current share template and let you figure out a plain theme option from there
- add it as a completely separate template file and leave figuring out the enablement via attributes or toggle or something to you
- if you already have a good idea of how you would want this to be selected/enabled by the user I can implement that too as part of the PR

**Reply by r4j4-s** [reply]
@zerebos Any update on this? Or are you working on including your theme as one of the default theme options for share in TriliumNext?
